File: Programacion Modulo Compras/SAE-COMPRAS/CapaNegocio/NOrdenDetalles.vb
Imports System.Collections.Generic
Imports System.Data
Imports CapaDatos
Imports System.Linq
Imports System.Text
Imports System.Threading.Tasks

Public Class NOrdenDetalles
	Public Shared Function MostrarOrdenDetalles(parIdOrden As Integer) As DataTable
		Return New DOrdenDetalles().MostrarOrdenDetalles(parIdOrden)
	End Function

	Public Shared Function Insertar(parNombre_Producto As [String], parCantidad As Integer, parPrecio_Unitario As Decimal, parDescuento As Decimal) As [String]
		Dim ordenDetalles As New DOrdenDetalles()
		ordenDetalles.Nombre_Producto = parNombre_Producto
		ordenDetalles.Cantidad = parCantidad
		ordenDetalles.Precio_Unitario = parPrecio_Unitario
		ordenDetalles.Descuento = parDescuento

		Return ordenDetalles.Insertar(ordenDetalles)
	End Function

	Public Shared Function Editar(parIdOrden As Integer, parNombre_Producto As [String], parCantidad As Integer, parPrecio_Unitario As Decimal, parDescuento As Decimal) As [String]
		Dim ordenDetalles As New DOrdenDetalles()
		ordenDetalles.Id_Orden = parIdOrden
		ordenDetalles.Nombre_Producto = parNombre_Producto
		ordenDetalles.Cantidad = parCantidad
		ordenDetalles.Precio_Unitario = parPrecio_Unitario
		ordenDetalles.Descuento = parDescuento

		Return ordenDetalles.Editar(ordenDetalles)
	End Function

End Class

File: Programacion Modulo Compras/SAE-COMPRAS/CapaNegocio/NOrdenDetalles.cs
using System;
using System.Collections.Generic;
using System.Data;
using CapaDatos;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace CapaNegocio
{
    public class NOrdenDetalles
    {
        public static DataTable MostrarOrdenDetalles(int parIdOrden)
        {
            return new DOrdenDetalles().MostrarOrdenDetalles(parIdOrden);
        }

        public static String Insertar(String parNombre_Producto, int parCantidad,
            decimal parPrecio_Unitario, decimal parDescuento)
        {
            DOrdenDetalles ordenDetalles = new DOrdenDetalles();
            ordenDetalles.Nombre_Producto = parNombre_Producto;
            ordenDetalles.Cantidad = parCantidad;
            ordenDetalles.Precio_Unitario = parPrecio_Unitario;
            ordenDetalles.Descuento = parDescuento;

            return ordenDetalles.Insertar(ordenDetalles);
        }

        public static String Editar(int parIdOrden, String parNombre_Producto, int parCantidad,
            decimal parPrecio_Unitario, decimal parDescuento)
        {
            DOrdenDetalles ordenDetalles = new DOrdenDetalles();
            ordenDetalles.Id_Orden = parIdOrden;
            ordenDetalles.Nombre_Producto = parNombre_Producto;
            ordenDetalles.Cantidad = parCantidad;
            ordenDetalles.Precio_Unitario = parPrecio_Unitario;
            ordenDetalles.Descuento = parDescuento;

            return ordenDetalles.Editar(ordenDetalles);
        }

    }
}
